### PR TITLE
topic/JANRAIN-5916/add-delimiter-to-csv

### PR DESCRIPTION
**Pull request by dsabo-JR**
This code adds the ability to have a delimiter in the utils.to_cs() method.

### CODE REVIEW

File: janrain_datalib/utils.py [hunk 1]
```diff
@@ -28,15 +28,16 @@ def to_json(item, compact=False):
         kwargs['separators'] = (',', ': ')
     return json.dumps(item, **kwargs)
 
-def to_csv(row):
+def to_csv(row, delim=None):
     """Convert a list of items to a CSV string.
 
     Example:
-    >>> to_csv(['first', '2nd,with,commas', 'lasty'])
+    #>>> to_csv(['first', '2nd,with,commas', 'lasty'])
```
Comment (Paulo-Janrain) [inline janrain_datalib/utils.py]: Why add a `#` here?  Is it special syntax for Python comments?

File: janrain_datalib/utils.py [hunk 2]
```diff
@@ -54,7 +55,14 @@ def to_csv(row):
             item = to_json(item, compact=True)
         new_row.append(item)
     output = io.StringIO()
-    writer = csv.writer(output)
+
+    # if the delimiter is set use it
+    if(delim):
```
Comment (Paulo-Janrain) [inline janrain_datalib/utils.py]: What if you had `delim=','` in the function declaration?  Then you wouldn't need the `if` statement?  I don't know if that's the same as not passing it though.

If you need to leave the `if` statement, I'd say remove the parentheses around `delim`.

File: janrain_datalib/utils.py (hunk 1 above)
Comment (dsabo-JR) [inline janrain_datalib/utils.py]: Had an error at one point because of that. ill fix it and re-push

File: janrain_datalib/utils.py (hunk 2 above)
Comment (dsabo-JR) [inline janrain_datalib/utils.py]: I had considered this but if someone passes in None it will replace the comma and the method will error

Comment (Paulo-Janrain) [inline janrain_datalib/utils.py]: Ah, got it.

File: janrain_datalib/utils.py [hunk 3]
```diff
@@ -54,7 +55,14 @@ def to_csv(row):
             item = to_json(item, compact=True)
         new_row.append(item)
     output = io.StringIO()
-    writer = csv.writer(output)
+
+    # if the delimiter is set use it
+    if delim :
+        writer = csv.writer(output, delimiter=delim)
+    # otherwise use the default comma delimiter
+    else:
```
Comment (ebeesonjanrain) [inline janrain_datalib/utils.py]: You don't need this conditional, just set the default for delim to a comma

File: setup.py [hunk 4]
```diff
@@ -18,7 +18,8 @@
         PACKAGE,
     ],
     install_requires=[
-        "janrain-python-api == 0.4.0",
+        "janrain-python-api",
+        "requests",
```
Comment (ebeesonjanrain) [inline setup.py]: this shouldn't be neccessary - janrain-python-api has requests as a dependency

Comment (dsabo-JR) [inline setup.py]: Reverted this back to the original setup.py

File: setup.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
         PACKAGE,
     ],
     install_requires=[
-        "janrain-python-api == 0.4.0",
+         "janrain-python-api == 0.4.0",
```
Comment (ebeesonjanrain) [inline setup.py]: remove extra space

File: janrain_datalib/utils.py (hunk 3 above)
Comment (ebeesonjanrain) [inline janrain_datalib/utils.py]: after discussion this makes sense

File: setup.py (hunk 5 above)
Comment (dsabo-JR) [inline setup.py]: Fixed

File: janrain_datalib/utils.py [hunk 6]
```diff
@@ -28,7 +28,7 @@ def to_json(item, compact=False):
         kwargs['separators'] = (',', ': ')
     return json.dumps(item, **kwargs)
 
-def to_csv(row):
+def to_csv(row, delim=None):
```
Comment (cameron-white-janrain) [inline janrain_datalib/utils.py]: Why `delim` when you could use the whole word `delimiter`?

File: janrain_datalib/utils.py [hunk 7]
```diff
@@ -37,6 +37,7 @@ def to_csv(row):
 
     Args:
         row: list of items to be formatted as CSV
+        delim: the delimiter to use for the CSV output
```
Comment (ebeesonjanrain) [inline janrain_datalib/utils.py]: change docs to reflect parameter name change

**Comment by ebeesonjanrain** [comment]
not sure if there's anything in jira to do for this release, but as far as code - we just increment the version, create a source dist tarball, and then submit it to pypi

**Pull request merged**